ibis/expr/types/joins.py

```diff
@@ -46,34 +46,28 @@ def disambiguate_fields(how, left_fields, right_fields, lname, rname):
     return fields, collisions
 
 
-def dereference_targets(chain):
-    yield chain.first
-    for join in chain.rest:
-        if join.how not in ("semi", "anti"):
-            yield join.table
-
-
 def dereference_mapping_left(chain):
-    rels = dereference_targets(chain)
+    # construct the list of join table we wish to dereference fields to
+    rels = [chain.first]
+    for link in chain.rest:
+        if link.how not in ("semi", "anti"):
+            rels.append(link.table)
```

Comment:
```suggestion
    rels.extend(link.table for link in chain.rest if link.how not in ("semi", "anti"))
```

Just a stylistic preference, in no way a blocker.